src/commcare_cloud/ansible/deploy_common.yml

```diff
@@ -93,7 +93,26 @@
     tags:
       - dns
       - after-reboot
+  
+  - name: Checks existing resolv.conf
+    stat:
+      path: /run/systemd/resolve/resolv.conf
+    register: resolvconf
+    tags:
+      - dns
+      - after-reboot
 
+  - name: Updating resolv.conf symbolic link 
+    when: ansible_distribution == "Ubuntu" and resolvconf.stat.exists
+    become: yes
+    file:
+      src: /run/systemd/resolve/resolv.conf
+      dest: /etc/resolv.conf
+      state: link
+    tags:
+      - dns
+      - after-reboot
```

Comment:
In the task above this, we make sure that the `nameservers` appear in `/etc/resolv.conf`, but then we overwrite /etc/resolve.conf in this task with a link to `/run/systemd/resolve/resolv.conf`. Where does `/run/systemd/resolve/resolv.conf` come from? Does the Ubuntu distribution for AWS include aws nameservers that will resolve private dns entries we specify in route53?

Reply:
In this case, we are changing  the symbolic link of /etc/resolv.conf. Presently, This file is mapping to **_/run/systemd/resolve/stub-resolv.conf_** which is containing the 127.0.0.53 DNS stub as the only DNS server [nameserver] and maintained by _**systemd-resolved**_ system service. This stub DNS-nameserver is unable to resolve AWS Route-53 Private hosted-zone records.

To resolve the private hosted-zones also, we are changing the symbolic link to **_/run/systemd/resolve/resolv.conf_** which contains information about all known DNS nameservers [ AmazonProvidedDNS ] and maintained by _**systemd-resolved**_ system service. This known DNS nameserver is able to resolve AWS Route-53 Private hosted-zone records also.

Reply:
OK thanks that makes sense. the task above this edits /etc/resolv.conf to include whatever we put in the `nameservers` variable. I think on AWS environments we don't need to specify custom `nameservers` (india does right now, but you can remove that; I'm pretty sure it's not necessary anymore). But third party environments (this is open source software and we encourage third parties running commcare to use it as well) may depend on `nameservers` to use appropriate DNS nameservers for their area or to override whatever their ISP's default is. Can you think through how this would work in that case? (I think as it's written we'd add nameservers to `/run/systemd/resolve/stub-resolv.conf` and then use `/run/systemd/resolve/resolv.conf`, which defeats the purpose of adding nameservers in the first place, right?

Reply:
> Presently, This file is mapping to **/run/systemd/resolve/stub-resolv.conf**

Is this the system "default state", or is this a modified system state (i.e. overridden by some custom configuration)?

> we are changing the symbolic link to **/run/systemd/resolve/resolv.conf**

Why? What populates this file, and why are its contents different from `/run/systemd/resolve/stub-resolv.conf`?

Is there `systemd-resolved.service` documentation that outlines why/how to do this sort of thing?